services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/s3express/S3ExpressIdentityCache.java

```diff
@@ -75,28 +76,42 @@ private CachedS3ExpressCredentials getCachedCredentials(S3ExpressIdentityKey key
                                          .build();
     }
 
-    //TODO (s3express) user experience and error messaging when calls fail
     SessionCredentials getCredentials(S3ExpressIdentityKey key, IdentityProvider<AwsCredentialsIdentity> provider) {
         SdkClient client = key.client();
         String bucket = key.bucket();
+        SdkServiceClientConfiguration serviceClientConfiguration = client.serviceClientConfiguration();
 
         if (client instanceof S3AsyncClient) {
             // TODO (s3express) don't join here
-            return ((S3AsyncClient) client).createSession(createSessionRequest(bucket, provider)).join().credentials();
+            return ((S3AsyncClient) client).createSession(createSessionRequest(bucket, provider, serviceClientConfiguration))
+                                           .join()
+                                           .credentials();
         }
         if (client instanceof S3Client) {
-            return ((S3Client) client).createSession(createSessionRequest(bucket, provider)).credentials();
+            return ((S3Client) client).createSession(createSessionRequest(bucket, provider, serviceClientConfiguration))
+                                      .credentials();
         }
         throw new UnsupportedOperationException("SdkClient must be either an S3Client or an S3AsyncClient, but was " +
                                                 client.getClass());
     }
 
-    private static Consumer<CreateSessionRequest.Builder>
+    private static CreateSessionRequest
             createSessionRequest(String bucket,
-                                 IdentityProvider<AwsCredentialsIdentity> provider) {
-        return r -> r.bucket(bucket)
+                                 IdentityProvider<AwsCredentialsIdentity> provider,
+                                 SdkServiceClientConfiguration serviceClientConfiguration) {
+
+        Duration requestApiCallTimeout = clientSetTimeoutIfExists(serviceClientConfiguration).orElse(DEFAULT_API_CALL_TIMEOUT);
```

Comment:
This is fine for now, but we might want to propagate all expected configuration, not just client config.